packages/tokenbank_v1/contracts/base/BaseVault.sol

```diff
@@ -0,0 +1,185 @@
+
+    uint public _requirement;
+
+    modifier onlyFromExecute
+    {
+        require(msg.sender == address(this), "NOT_FROM_THIS_MODULE");
+        _;
+    }
+
+    constructor(
+        address[] memory owners,
+        uint      initialRequirement
+        )
+        public
+    {
+        require(owners.length > 0, "NULL_OWNERS");
+        require(owners.length <= MAX_OWNERS, "TOO_MANY_OWNERS");
+
+        for (uint i = 0; i < owners.length; i++) {
+            address owner = owners[i];
+            require(owner != address(0), "ZERO_ADDRESS");
+            addAddressToSet(OWNERS, owner, true);
+        }
+        require(
+            initialRequirement > 0 &&
+            initialRequirement <= owners.length,
+            "INVALID_REQUIREMENT"
+        );
+        _requirement = initialRequirement;
+    }
+
+    function execute(
+        address   target,
+        uint      value,
+        bytes     calldata data,
+        address[] calldata signers,
+        bytes     calldata signatures
+        )
+        external
+    {
+        require(signers.length >= _requirement, "NEED_MORE_SIGNATURES");
+        bytes32 signHash = getSignHash(
+            address(this), // from
+            target,
+            value,
+            data
+        );
+        signHash.verifySignatures(signers, signatures);
+        // solium-disable-next-line security/no-call-value
+        (bool success, ) = target.call.value(value)(data);
+        emit Executed(target, value, data, success);
+    }
+
+    function getSignHash(
+        address from,
+        address to,
+        uint256 value,
+        bytes   memory data
+        )
+        public
+        pure
+        returns (bytes32)
+    {
+        bytes32 hash = keccak256(
+            abi.encodePacked(
+                byte(0x19),
+                byte(0),
+                from,
+                to,
+                value,
+                keccak256(data)
+                )
+            );
+
+        return keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash));
+    }
+
+    function addOwner(address owner)
+        external
+        onlyFromExecute
+    {
+        addAddressToSet(OWNERS, owner, true);
+        emit OwnerAdded(owner);
+    }
+
+    function removeOwner(address owner)
+        external
+        onlyFromExecute
+    {
+        uint count = numOwners();
+        require(count > 1, "PROHIBITED");
+
+        if (_requirement == count) {
+            _requirement -= 1;
+        }
+
+        removeAddressFromSet(OWNERS, owner);
+        emit OwnerRemoved(owner);
+    }
+
+    function changeRequirement(uint256 newRequirement)
+        external
+        onlyFromExecute
+    {
+        require(
+            newRequirement > 0 &&
+            newRequirement <= numOwners(),
+            "INVALID_REQUIREMENT"
+        );
+        _requirement = newRequirement;
+        emit RequirementChanged(_requirement);
+    }
+
+    function requirement()
```

Comment:
`_requirement` -> `requirement` and solidity will automatically generate a `view`method for it if it's public (which `_requirement` is). Or is there another reason to do it like this?

Reply:
no it's not working as `requirement()` is an interface method.